Bottom: 72fccdc018c6f9ede117dd65d21b625ef83e3c6a
Top:    936dc641ef8c19b0c9639d402a531e0c85cc05e9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-10 02:52:31 +0000

Refresh of convet-some-safe-usage-of-myst

---

diff --git a/src/create.c b/src/create.c
index a1c30bb..735f20f 100644
--- a/src/create.c
+++ b/src/create.c
@@ -1011,8 +1011,8 @@ CreateNewAttribute (AttributeListTypePtr list, char *name, char *value)
 			      list->Max * sizeof (AttributeType),
 			      "CreateNewAttribute");
     }
-  list->List[list->Number].name = strdup (name);
-  list->List[list->Number].value = strdup (value);
+  list->List[list->Number].name = MyStrdup (name, "CreateNewAttribute");
+  list->List[list->Number].value = MyStrdup (value, "CreateNewAttribute");
   list->Number++;
   return &list->List[list->Number - 1];
 }
diff --git a/src/misc.c b/src/misc.c
index 40c6460..9ab9f53 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1920,7 +1920,7 @@ AttributePutToList (AttributeListType *list, const char *name, const char *value
 
   /* Now add the new attribute.  */
   i = list->Number;
-  list->List[i].name = strdup (name);
+  list->List[i].name = MyStrdup (name, "AttributePutToList");
   list->List[i].value = MyStrdup (value, "AttributePutToList");
   list->Number ++;
   return 0;
